### PR TITLE
feat: add afterScreenshot option

### LINKED ISSUE #395: Add hooks
### Mode

Other

### Feature description

I would like to suggest adding a new feature to the project configuration.

The feature request is to add hooks. Under hook I mean some subscription on events that are happening under the hood and executing some user logic. Example: [Nuxt hooks](https://nuxt.com/docs/guide/going-further/hooks)

Suggested configuration:

```ts
import type { CustomProjectConfig } from "lost-pixel"

export const config: CustomProjectConfig = {
  // ...
  hooks: {
    "<hook-name>": async function (args) {
      await doSomethingWithArgs(args)
    },
  },
}
```

The expected process flow:

1. executing lib logic
2. executing user hook logic (the process is stopped until the hook handler is not resolved)
3. executing lib logic
4. ....
5. finish the process

**My case**: I would like to edit generated screenshots before making a comparison. Usually, I use visual comparison for checking the state of reusable components, most of the time they are much bigger than the viewport. So I wrote a script that executes on provided images and can "trim" empty spaces (pixels with white color, from all sides until a row or column with at least 1 pixel with non-white color). I want to execute this script after screenshots were generated, but before the comparison started.

Would it be useful to add such a feature (hooks) to your library?
If the suggested feature is out of the library's scope, could you share please any other way how I can achieve the desired behavior?

### PR DESCRIPTION
**Pull request by levchak0910**
All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/lost-pixel/lost-pixel/blob/main/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lost-pixel/lost-pixel/pulls) for the same update/change?

### New Feature Submissions:

1. [x] Was the new feature discussed with core maintainers?
2. [x] Does your submission pass tests?
3. [x] Have you lint your code locally before submission?
4. [x] Does the change require an update in documentation? Have you updated the documentation in case it's true?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully run tests with your changes locally?
* [x] Does the change require an update in documentation? Have you updated the documentation in case it's true?

closes #395

### CODE REVIEW

**Comment by d-ivashchuk** [comment]
@levchak0910 looks good from my side, I will let @chriskalmar take a brief look to see if he might have any good ideas to extend this. 

really nice work all together including adding new vue histoire example!

File: src/config.ts [hunk 1]
```diff
@@ -386,6 +386,19 @@ const BaseConfigSchema = z.object({
...
+
+  /**
+   * Perform actions after screenshot was shot
+   */
+  afterScreenshot: z
+    .function()
+    .args(
+      z.object({
+        shotPath: z.string(),
+      }),
```
Comment (chriskalmar) [inline src/config.ts:398]: To keep it more flexible in the future, I'd suggest changing that to:
```
.args(z.custom<Page>(), StoryLikeSchema)
```

and extend the `StoryLikeSchema` with `filePathBaseline`, `filePathCurrent`, and `filePathDifference` (all optional props).

Comment (levchak0910) [reply src/config.ts]: ready

**Review by chriskalmar — APPROVED** [review]
@levchak0910 Thank you for your contribution

**Pull request merged**